libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -542,8 +537,11 @@ proc send*(
       sendMetrics(msg)
       encodeRpcMsg(msg, anonymize)
 
-  if encoded.len > p.maxMessageSize and msg.messages.len > 1:
-    for encodedSplitMsg in splitRPCMsg(p, msg, p.maxMessageSize, anonymize):
+  # Messages should not exceed 90% of maxMessageSize. Guessing 10% protobuf overhead.
+  let maxEncodedMsgSize = (p.maxMessageSize * 90) div 100 
```

Comment:
nit: I think `maxEncodedMsgSize` should be part of `PubSubPeer`, initialized when it's created using a `const` that says how much of the messageSize should be the `maxEncodedMsgSize` (here, that const would be `0.9`)